persistence/test/actor_test.go

```diff
@@ -37,3 +40,79 @@ func TestGetAllStakedActors(t *testing.T) {
 	require.Equal(t, genesisStateNumApplications, actualApplications)
 	require.Equal(t, genesisStateNumFishermen, actualFishermen)
 }
+
+func TestGetValidatorSet_Hash(t *testing.T) {
+	expectedHashes := []string{
+		"5831e3e6a8d3beda0adb6027126a4bc3b0181836eebcc45a83dc2970ee9b4468",
+		"1f6faa8782a4608341fef83ee72c9e9cd3f96e042f2ddfdeebda8d971bb2ac13",
+	}
+
+	db := NewTestPostgresContext(t, 0)
+	currValSet, err := db.GetValidatorSet(0)
+	require.NoError(t, err)
+	currValSetHash := hashValSet(t, currValSet)
+	require.Equal(t, expectedHashes[0], currValSetHash)
+
+	currHeight := int64(1)
+	db.Height = currHeight
+
+	currValSet, err = db.GetValidatorSet(currHeight - 1)
+	require.NoError(t, err)
+	currValSetHash = hashValSet(t, currValSet)
+	nextValSet, err := db.GetValidatorSet(currHeight)
+	require.NoError(t, err)
+	nextValSetHash := hashValSet(t, nextValSet)
+
+	require.Equal(t, expectedHashes[0], currValSetHash)
+	require.Equal(t, expectedHashes[0], nextValSetHash)
+
+	// ensure hash remains the same with no changes
+	currHeight = int64(2)
+	db.Height = currHeight
+
+	currValSet, err = db.GetValidatorSet(currHeight - 1)
+	require.NoError(t, err)
+	currValSetHash = hashValSet(t, currValSet)
+	nextValSet, err = db.GetValidatorSet(currHeight)
+	require.NoError(t, err)
+	nextValSetHash = hashValSet(t, nextValSet)
+
+	require.Equal(t, expectedHashes[0], currValSetHash)
+	require.Equal(t, expectedHashes[0], nextValSetHash)
+
+	// ensure hash changes with changes
+	currHeight = int64(3)
+	db.Height = currHeight
+
+	err = db.InsertValidator(
+		[]byte("address"),
+		[]byte("publickey"),
+		[]byte("output"),
+		false, 0,
+		"serviceurl",
+		"1000000000",
+		0,
+		0,
+	)
+	require.NoError(t, err)
+
+	currValSet, err = db.GetValidatorSet(currHeight - 1)
```

Comment:
Can you also update the test to insure the validators are sorted by address?

Reply:
So the validators are sorted by address and height this means they could look like this:
"aaaaaaa"(height10)
"ccccccc" (height9)
"bbbbbb"(height9)
"zzzzzzz"(height1)
"ddddddd"(height1)
What is important is that although the ordering may not be lexigraphically ordered by address alone it is still deterministic by height. This means the hash will be consistent by height. I think testing they are lexigraphically ordered only passes at genesis and afterwards will not alway be the case

Reply:
My request is to update the test to verify this explicitly:

For example:
```
require.Equal(t, "aaaaa", valSet[0].address) // height10
require.Equal(t, "cccc", valSet[0].address) // height9
require.Equal(t, "bbb", valSet[0].address) // height8
```

Makes it easier for future maintainers of the code and also increase the surface area of bugs

Reply:
My bad I actually misunderstood the query. They are sorted by address ASC alone as the select is ON DISTINCT it means even if the address is at multiple heights only the most recent is chosen but results are still sorted by address ASC. Updated to test for this.